### PR TITLE
Add slurm-stats

### PR DESCRIPTION
**Pull request by jovial**
Trying to group tools with similar theme to prevent proliferation of repositories.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -36,3 +40,93 @@ Non-openstack nodes will ignore the reason, and just do a regular reboot.
...
+
+Command-line tool which transforms sacct output into a form that is more
+amenable for importing into elasticsearch/loki.
+
+It uses slurm's sacct to extract job stats of finished jobs. It stores the
+last timestamp in a file called "lasttimestamp". This means the tool can fetch all
+jobs since the last time stamp. If there is no stored timestime it defaults to
+fetching all jobs since midnight.
+
+Note: the default OpenHPC install doesn't enable the accounting service. It appears
```
Comment (sjpb) [inline README.rst:55]: I'm not sure this is quite right - it does enable accounting but with `AccountingStorageType` defaulted to ` accounting_storage/filetxt` and it appears `accounting_storage/slurmdbd` is required for this.

**Review by sjpb — CHANGES_REQUESTED** [review]
Not sure this review has added much value! Things like slurm commands/parsing looks ok though.

File: slurm_openstack_tools/sacct.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    # Work out starttime and endtime
+    now = datetime.datetime.utcnow()
+    end_str = now.strftime(SLURM_DATE_FORMAT)
+
+    try:
+        with open(TIMESTAMP_FILE) as f:
+            start_str = f.read()
+    except FileNotFoundError:
+        # Default to start of today
+        start_str = "00:00:00"
```
Comment (JohnGarbutt) [inline slurm_openstack_tools/sacct.py:29]: I think this is wrong, we should have start_str = ""

File: slurm_openstack_tools/sacct.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    args += ["--starttime", start_str]
```
Comment (JohnGarbutt) [inline slurm_openstack_tools/sacct.py:31]: I think here we need:

if start_str:
  args += ["--starttime", start_str]

Comment (jovial) [reply slurm_openstack_tools/sacct.py:33]: Went for collecting jobs from up to a year ago. Using a date too far in the past resulted in no output :-/

**Comment by jovial** [comment]
Merging as all those open PRs make it hard to test anything.

**Pull request merged**